python/pyspark/ml/executor_globals.py

```diff
@@ -0,0 +1,24 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Module to hold globals for python processes on executors
+from typing import Callable, Optional
+from uuid import UUID
+
+
+model_uuid: Optional[UUID] = None
+predict_fn: Optional[Callable] = None
```

Comment:
This is not the correct way of building cache:
It is not thread safe. Many concurrent spark tasks (might comes from different spark job) might be spawn on executor side.

To cache model, you can copy mlflow code of:
https://github.com/mlflow/mlflow/blob/master/mlflow/pyfunc/spark_model_cache.py

Reply:
Right, updated the code.  Probably still need to figure out a scheme for cache invalidation for large models and/or frequent use.